### PR TITLE
VerificationError -> AssertionError

### LINKED ISSUE #1: Use AssertionError instead of VerificationError?
AssertionError is a bulitin exception type, and can make the writing of verifiers easier (also, no need for weird shenanigans with importing VerificationError from the module).

For verifiers, you could just do something like:

```
assert <the condition>, "The message"
```

which is a lot more readable than the current system.

I can make the change, just wanted to get this approved before doing so.

EDIT:

- Pasting this here for my own reference when I get around to implementing this: https://stackoverflow.com/questions/2380073/how-to-identify-what-function-call-raise-an-exception-in-python

### PR DESCRIPTION
**Pull request by sumnerevans**
Closes #1

#9 created as followup.

### CODE REVIEW

**Comment by sumnerevans** [comment]
I'm not entirely satisfied with this, as there's no backwards compatibility for this change. I'm curious what you think is the best approach for adding backwards compatibility? I don't want to just add another `except` block for `verif_mod.VerificationError`, as `VerificationError` may not exist in the new verifiers.

**Comment by jackrosenthal** [comment]
Can `VerifierModule` define an attribute for the verification error, which is set to `AssertionError` if the module does not provide `VerificationError`?

File: algobowl/controllers/competition.py [hunk 1]
```diff
@@ -301,6 +301,8 @@ def new_gt(rankings_entry):
     @require(has_permission('admin'))
     def reverify(self):
         verif_mod = self.competition.output_verifier.module
+        if not getattr(verif_mod, 'VerificationError'):
```
Comment (jackrosenthal) [inline algobowl/controllers/competition.py:304]: hasattr

File: algobowl/controllers/group.py [hunk 2]
```diff
@@ -68,9 +68,16 @@ def input_upload(self, input_upload=None, team_name=None):
             if len(contents) > 1E6:
                 abort(400, "Your input exceeds the maxiumum size.")
             verif_mod = self.group.competition.input_verifier.module
+            if not getattr(verif_mod, 'VerificationError'):
```
Comment (jackrosenthal) [inline algobowl/controllers/group.py:71]: ditto

File: algobowl/controllers/group.py [hunk 3]
```diff
@@ -68,9 +68,16 @@ def input_upload(self, input_upload=None, team_name=None):
...
+                verif_mod.VerificationError = AssertionError
 
             try:
                 verif_mod.verify(StringIO(contents))
+            except verif_mod.VerificationError as e:
+                flash('Your input has been rejected for the following reason: '
+                      '{}. Please correct and try uploading again.'.format(e),
+                      'danger')
+                redirect(self.base_url)
             except AssertionError as e:
```
Comment (jackrosenthal) [inline algobowl/controllers/group.py:81]: drop

File: algobowl/controllers/group.py [hunk 4]
```diff
@@ -159,8 +166,12 @@ def submit_output(self, to_group, output_file=None):
                         score=score, original=comp.output_upload_open)
 
         verif_mod = self.group.competition.output_verifier.module
+        if not getattr(verif_mod, 'VerificationError'):
+            verif_mod.VerificationError = AssertionError
         try:
             verif_mod.verify(to_group.input.data.file, StringIO(contents))
+        except verif_mod.VerificationError as e:
+            output.ground_truth = VerificationStatus.rejected
         except AssertionError:
```
Comment (jackrosenthal) [inline algobowl/controllers/group.py:175]: ditto

**Comment by jackrosenthal** [comment]
Squish the commits and add BUG=#8 to the bottom of the commit message.

**Comment by jackrosenthal** [comment]
not needed anymore with the new problem format

**Pull request closed without merging**